### PR TITLE
MM-24285 Use FastImage instead of Image

### LINKED ISSUE #4190: Custom emoji images not shown
#### Summary
Custom emoji images are not shown despite the registered emoji are available on the list of emoji. The images are shown on the desktop app properly.

#### Environment Information

- Device Name: SH-M12
- OS Version: Android 9
- Mattermost App Version: 1.30.0
- Mattermost Server Version: 5.21.0

##### Case 1

See custom emoji in someone's message.

##### Case 2

See custom emoji reaction on someone's message.

##### Case 3

1. Make the list of emoji pop up by putting : (colon) at the input text field.
1. Find your custom emoji by name.

#### Expected behavior
Images of custom emoji are shown.

#### Observed behavior
Images of custom emoji are blank.

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Currently we are using a combination of `Image` and `FastImage`, with this PR we start using `FastImage` for all images that are loading through the network to make use of the aggressive cache provided by fast image boosting performance when loading already cached images.

Re-added missing placeholder while the image file attachments are being loaded.

Test areas:
* Image attachments (local and over the network)
* YouTube thumbnails
* OpenGraph images
* Message attachments images and author images
* Markdown Images
* Image link previews
* Image sharing via Share Extensions

Side note: There is unrelated code changes due to the fact that an ESLint rule was changed to disallow unused vars.

Fixes #4190 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-24285

### CODE REVIEW

File: app/screens/image_preview/image_preview.js [hunk 1]
```diff
@@ -375,11 +375,12 @@ export default class ImagePreview extends PureComponent {
             const flattenStyle = StyleSheet.flatten(style);
             const calculatedDimensions = calculateDimensions(height, width, deviceWidth, deviceHeight - statusBar);
             const imageStyle = {...flattenStyle, ...calculatedDimensions};
+            const src = {...source, cache: FastImage.cacheControl.cacheOnly};
```
Comment (chuttam) [inline app/screens/image_preview/image_preview.js:378]: Why not get from original source if (somehow) cached version doesn't exist?

File: app/components/markdown/markdown_image/markdown_image.js [hunk 2]
```diff
@@ -256,20 +215,20 @@ export default class MarkdownImage extends React.PureComponent {
...
                         onPress={this.handlePreviewImage}
-                        style={{width, height}}
+                        style={{width: 320, height}}
                     >
                         <ProgressiveImage
                             ref={this.setImageRef}
                             defaultSource={source}
                             resizeMode='contain'
-                            style={{width, height}}
+                            style={{width: 323, height}}
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/components/markdown/markdown_image/markdown_image.js:224]: 3 px to align with other images... that's all

**Comment by saturninoabril** [comment]
@enahum Tested those areas and all passed except for image link preview:
- Posting `https://user-images.githubusercontent.com/29700158/80384256-9c532280-88df-11ea-9059-aab558a916d9.png` did not show preview
Screen Shot 2020-04-28 at 1 14 32 AM
- Posting `Inline-style: 
alt text` rendered the image on center with border.
Screen Shot 2020-04-28 at 1 29 10 AM

**Comment by enahum** [comment]
Thanks Saturn, I’ll look into both

**Comment by enahum** [comment]
@saturninoabril ready for re-review

File: app/screens/image_preview/image_preview.js (hunk 1 above)
Comment (enahum) [reply app/screens/image_preview/image_preview.js:378]: if cache version does not exist at this point we have an error in logic elsewhere.. this is for the gallery view and I do expect to have it cached by this point

**Comment by enahum** [comment]
manually cherry-pick as 8e314022ca81680a7129c018cb2eaaf13ad7800b

**Pull request merged**